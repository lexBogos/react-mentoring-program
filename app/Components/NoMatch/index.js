import React from 'react';
import {
    useLocation
  } from "react-router-dom";
import './index.scss';


const NoMatch = () => {
    let location = useLocation();
    return <>
        <h1>No match for <code>{location.pathname}</code> (Error 404)</h1>
     </>
}


export default NoMatch;