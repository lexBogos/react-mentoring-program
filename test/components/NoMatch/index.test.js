import React from 'react';
import NoMatch from '../../../app/Components/NoMatch/index';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme,{ mount } from 'enzyme';
import '@testing-library/jest-dom/extend-expect';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('react-router-dom', () => ({
    useLocation: jest.fn().mockReturnValue({
      pathname: '/another-route',
      search: '',
      hash: '',
      state: null,
      key: '5nvxpbdafa',
    }),
}));

describe('NoMatch component', () => {

        it('render NoMatch component toMatchSnapshot', () => {
            expect(mount(<NoMatch/>).debug()).toMatchSnapshot();
        })

})