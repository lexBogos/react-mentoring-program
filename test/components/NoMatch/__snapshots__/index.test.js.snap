// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoMatch component render NoMatch component toMatchSnapshot 1`] = `
"<NoMatch>
  <h1>
    No match for 
    <code>
      /another-route
    </code>
     (Error 404)
  </h1>
</NoMatch>"
`;
